src/traits.rs

```diff
@@ -202,14 +202,17 @@ pub trait Split<Rhs = Self> {
     fn split(&self) -> (Self::Output, Self::Output);
 }
 
-/// Encoding support.
-pub trait Encoding: Sized {
+/// Integers whose representation takes a bounded amount of space.
+pub trait Bounded {
```

Comment:
Separating out a trait for the sizes from `Encoding` seems good although I'm not sure about the name itself, but of course `Sized` is already taken by `core`.

It seems `num-traits` uses [`Bounded` to represent min and maximum values](https://docs.rs/num-traits/latest/num_traits/bounds/trait.Bounded.html) which seems like a better fit for that name.

If I can't think of anything better, `Bounded` seems OK.

Reply:
> It seems `num-traits` uses `Bounded` to represent min and maximum values which seems like a better fit for that name.

Yes, that's where I took the name from. Those values can be potentially added to the trait too, but I didn't want to do it in this PR. For unsigned integers `BITS` is just a different representation of `MAX`, anyway. I think `BITS` should be separate from the `Encoding` trait (since we already have `.bits()` outside of it, which is connected to `BITS`), but as for the name of the new trait, I'm open to suggestions. 

The conceptual problem here is that `pow_bounded_exp()` makes sense only for unsigned integer exponents; so technically "Bounded" is too loose of a restriction and will have to be changed if signed integers are introduced. The "theoretical" bound would be `Unsigned + Integer + Bits", with the last two possibly implied by `Unsigned`.

Reply:
Alright, let's go with `Bounded` for now and we can revisit it later if need be